### PR TITLE
Fix undef ref in lkj_chol sampling

### PR DESCRIPTION
**Pull request by chriselrod**

### CODE REVIEW

**Comment by sethaxen** [comment]
Thanks, can you add a test that would have failed before this PR but now passes?

File: test/cholesky/lkjcholesky.jl [hunk 1]
```diff
@@ -181,6 +181,7 @@ using FiniteDifferences
         nkstests = 4 # use for appropriate Bonferroni correction for KS test
 
         @testset "rand" begin
+            @test only(rand(LKJCholesky(1, 0.5)).L) == 1.0
```
Comment (devmotion) [inline test/cholesky/lkjcholesky.jl:184]: `only` was introduced in 1.4 IIRC.
```suggestion
            @test rand(LKJCholesky(1, 0.5)) == ones(1, 1)
```

File: test/cholesky/lkjcholesky.jl [hunk 2]
```diff
@@ -181,6 +181,7 @@ using FiniteDifferences
         nkstests = 4 # use for appropriate Bonferroni correction for KS test
 
         @testset "rand" begin
+            @test rand(LKJCholesky(1, 0.5)) == ones(1, 1)
```
Comment (sethaxen) [inline test/cholesky/lkjcholesky.jl:184]:
```suggestion
            @test rand(LKJCholesky(1, 0.5)) == Cholesky(ones(1, 1), 'U', 0)
```

File: test/cholesky/lkjcholesky.jl [hunk 3]
```diff
@@ -181,6 +181,7 @@ using FiniteDifferences
         nkstests = 4 # use for appropriate Bonferroni correction for KS test
 
         @testset "rand" begin
+            @test rand(LKJCholesky(1, 0.5)) == Cholesky(ones(1, 1), 'U', 0)
```
Comment (sethaxen) [inline test/cholesky/lkjcholesky.jl:184]: Sorry, forgot the default is L
```suggestion
            @test rand(LKJCholesky(1, 0.5)) == Cholesky(ones(1, 1), 'L', 0)
```

Comment (devmotion) [reply test/cholesky/lkjcholesky.jl:184]: Maybe the safer option would be
```suggestion
            @test rand(LKJCholesky(1, 0.5)).factors == ones(1, 1)
```
?

**Review by devmotion — APPROVED** [review]
Thank you 🙂

**Pull request merged**